src/app/handlers/flightsql.rs

```diff
@@ -66,74 +66,74 @@ pub fn normal_mode_handler(app: &mut App, key: KeyEvent) {
-                                            // be some sort of paging logic
-                                            // see https://github.com/datafusion-contrib/datafusion-tui/pull/133#discussion_r1756680874
-                                            // while let Some(maybe_batch) = stream.next().await {
-                                            if let Some(maybe_batch) = stream.next().await {
-                                                match maybe_batch {
-                                                    Ok(batch) => {
-                                                        info!("Batch rows: {}", batch.num_rows());
-                                                        batches.push(batch);
-                                                    }
-                                                    Err(e) => {
-                                                        error!("Error getting batch: {:?}", e);
-                                                        let elapsed = start.elapsed();
-                                                        query.set_error(Some(e.to_string()));
-                                                        query.set_execution_time(elapsed);
-                                                    }
-                                                }
-                                            }
-                                            let elapsed = start.elapsed();
-                                            let rows: usize =
-                                                batches.iter().map(|r| r.num_rows()).sum();
-                                            query.set_results(Some(batches));
-                                            query.set_num_rows(Some(rows));
-                                            query.set_execution_time(elapsed);
-                                        }
-                                        Err(e) => {
-                                            error!("Error getting response: {:?}", e);
-                                            let elapsed = start.elapsed();
-                                            query.set_error(Some(e.to_string()));
-                                            query.set_execution_time(elapsed);
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                        Err(e) => {
-                            error!("Error getting response: {:?}", e);
-                            let elapsed = start.elapsed();
-                            query.set_error(Some(e.to_string()));
-                            query.set_execution_time(elapsed);
-                        }
-                    }
-                }
-
-                let _ = _event_tx.send(AppEvent::FlightSQLQueryResult(query));
-            });
-        }
+        // KeyCode::Enter => {
```

Comment:
Going to come back to this in a follow on PR for paginating FlightSQL

Reply:
I wonder if we can somehow hide it behind an interface (so the pagination code for FlightSQL and local are the same) 🤔

Reply:
I agree that would be nice.  I want to finalize the functionality for flightsql first - then with both features complete i think it will be easier to reason about what the appropriate interface should be.